extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -466,18 +466,12 @@ export class InstrumentationService {
    */
   matchesSelector_(el, selector) {
     try {
-      const matcher = el.matches ||
-          el.webkitMatchesSelector ||
-          el.mozMatchesSelector ||
-          el.msMatchesSelector ||
-          el.oMatchesSelector;
-      if (matcher) {
-        return matcher.call(el, selector);
-      }
-      const matches = el.ownerDocument.querySelectorAll(selector);
-      let i = matches.length;
-      while (i-- > 0 && matches.item(i) != el) {};
-      return i > -1;
+      return matches(el, selector, () => {
+        const matches = el.ownerDocument.querySelectorAll(selector);
```

Comment:
Shouldn't this part simply go inside `matches` utility? This is a polyfill, right? For a rare case of browsers that still need it, this seems totally justifiable.